stl/inc/xstring

```diff
@@ -2569,37 +2547,43 @@ private:
             return;
         }
 
-        // We need to check whether we have a misaligned SSO buffer because of the proxy in `_Container_base` (e.g. x86)
+        // When SSO is active, the buffer may not be aligned on an ASan shadow memory boundary;
         if constexpr (_Memcpy_val_offset % _Asan_granularity != 0) {
-            const uintptr_t _Alignment_offset = reinterpret_cast<uintptr_t>(_Ptr) & (_Asan_granularity - 1);
-            if (_Alignment_offset != 0 && _Capacity == _BUF_SIZE - 1) {
-                return;
+            // The alignment can be incorrect on this platform and for this type
+            if (_Capacity == _BUF_SIZE - 1) {
+                // This string is in SSO mode, so we need to check if the buffer happens to be aligned or not
+                const uintptr_t _Buffer_alignment = reinterpret_cast<uintptr_t>(_First) & (_Asan_granularity - 1);
+                if (_Buffer_alignment != 0) {
+                    // NOTE TO REVIEWER: shouldn't this do the `_Aligned_first` thing that's done when in large string mode?

```

Comment:
Maybe the prior intent was to throw our hands in the air, and not try to ASan annotate when the SSO us unaligned?